src/client/sandbox/event/drag-and-drop/data-transfer.js

```diff
@@ -0,0 +1,314 @@
+// NOTE: Although DataTransfer interface has a constructor, it is not possible to
+// create a useful DataTransfer object from script, since DataTransfer objects have a
+// processing and security model that is coordinated by the browser during drag-and-drops.
+// So we have to create a mock for it to use it in drag-and-drop events
+import { setTimeout } from '../../native-methods';
+
```

Comment:
Add links to specfication for each entry:
https://html.spec.whatwg.org/multipage/interaction.html#dom-datatransfer-dropeffect
https://html.spec.whatwg.org/multipage/interaction.html#dom-datatransfer-effectallowed
etc